addons/knowledge/static/src/js/knowledge_behavior_table_of_content.js

```diff
@@ -0,0 +1,214 @@
+            // Compute the 'depth' we want to display when increasing
+            if (depth !== previousDepth && heading.tagName === previousTag) {
+                // Same tag name as previous one -> use same depth
+                depth = previousDepth;
+            } else if (depth > previousDepth) {
+                // This is done only when changing tags, as 2 h4 in our previous example
+                // need to be at the same depth
+                if (heading.tagName !== previousTag) {
+                    // We only go max +1 depth at the time, meaning if a h4 follows a h1,
+                    // we want only one extra depth
+                    depth = previousDepth + 1;
+                }
+            } else if (depth < previousDepth) {
+                // When going down, it's different, we need to see if our current tree already
+                // has this tag at a certain depth, and use that
+                if (currentDepthByTag.hasOwnProperty(heading.tagName)) {
+                    depth = currentDepthByTag[heading.tagName];
+                }
+            }
+
+            previousTag = heading.tagName;
+            previousDepth = depth;
+
+            // going back to 0 depth, wipe-out the 'currentDepthByTag'
+            if (depth === 0) {
+                currentDepthByTag = {};
+            }
+            currentDepthByTag[heading.tagName] = depth;
+
+            return {
+                depth: depth,
+                index: index++,
+                name: heading.innerText,
+                tagName: heading.tagName,
+            }
+        });
+
+        const updatedToc = qweb.render('knowledge.knowledge_table_of_content', {
+            'headings': headingStructure
+        });
+        const knowledgeToCElement = this.anchor.getElementsByClassName('o_knowledge_toc_content');
+        if (knowledgeToCElement.length !== 0) {
+            knowledgeToCElement[0].innerHTML = updatedToc;
+        }
+
+        this.handler.editor.observerActive('knowledge_toc_update');
+    },
+
+    //--------------------------------------------------------------------------
+    // Table of content - HANDLERS
+    //--------------------------------------------------------------------------
+
+    /**
+     * Scroll through the view to display the matching heading.
+     * Adds a small highlight in/out animation using a class.
+     *
+     * @param {Event} event
+     */
+    _onTocLinkClick: function (event) {
+        event.preventDefault();
+        const headingIndex = parseInt(event.target.getAttribute('data-oe-nodeid'));
+        const targetHeading = fetchRelevantHeadings(this.handler.field)[headingIndex];
+        if (targetHeading){
+            targetHeading.scrollIntoView({
+                behavior: 'smooth',
+            });
+            targetHeading.setAttribute('highlight', true);
+            setTimeout(() => {
+                targetHeading.removeAttribute('highlight');
+            }, 2000);
```

Comment:
would be cool to add transition for the highlight. (and get another color ? and use rounded borders ? like the highlight borders in the TOC itself.)

EDIT:
Ah ! Found it ! You indeed added transition but it did not work.
The issue is that the transition is working only when setting the highlight. not when removing it.
You can achieve this doing the following : 
```
targetHeading.setAttribute('highlight', true);
            setTimeout(() => {
                targetHeading.setAttribute('highlight', false);
            }, 2000);
```
and in css:
```
            h1, h2, h3, h4, h5, h6 {
                &[highlight="true"] {
                    background-color: $o-knowledge-bg--active;
                }
                &[highlight="false"] {
                    background-color: transparent;
                    transition: background-color 0.5s ease;
                }
            }
```

Reply:
I have changed to use a class, which makes things a bit easier than a random attribute.
Also included rounded borders as suggested.

(Not sure about the color so I kept the same grey as hovered ToC link?)

 :artist: :it: